File: ChatGPT(예진)/Python (수정 완료)/Hangman_Game.py

import random


def choose_word():
    word_list = ["apple", "banana", "orange", "grape", "melon", "strawberry", "peach"]
    return random.choice(word_list)


def display_word(secret_word, guessed_letters):
    displayed_word = ""
    for letter in secret_word:
        if letter in guessed_letters:
            displayed_word += letter
        else:
            displayed_word += "_"
    return displayed_word


def hangman():
    secret_word = choose_word()
    guessed_letters = []
    attempts = 6

    print("Welcome to Hangman!")
    print("Try to guess the secret word.")
    print(display_word(secret_word, guessed_letters))

    while True:
        guess = input("Guess a letter: ").lower()

        if len(guess) != 1 or not guess.isalpha():
            print("Please enter a single letter.")
            continue

        if guess in guessed_letters:
            print("You've already guessed that letter.")
            continue

        guessed_letters.append(guess)

        if guess in secret_word:
            print("Correct!")
        else:
            print("Incorrect!")
            attempts -= 1

        displayed_word = display_word(secret_word, guessed_letters)
        print(displayed_word)

        if "_" not in displayed_word:
            print("Congratulations! You've guessed the word:", secret_word)
            break

        if attempts == 0:
            print("You've run out of attempts! The secret word was:", secret_word)
            break

        print("Attempts left:", attempts)


if __name__ == "__main__":
    hangman()

File: ChatGPT(예진)/Java (수정 완료)/Hangman_Game.java

import java.util.Scanner;
import java.util.Random;

public class Hangman_Game {
    public static String chooseWord() {
        String[] wordList = {"apple", "banana", "orange", "grape", "melon", "strawberry", "peach"};
        Random random = new Random();
        return wordList[random.nextInt(wordList.length)];
    }

    public static String displayWord(String secretWord, String guessedLetters) {
        StringBuilder displayedWord = new StringBuilder();
        for (int i = 0; i < secretWord.length(); i++) {
            char letter = secretWord.charAt(i);
            if (guessedLetters.contains(String.valueOf(letter))) {
                displayedWord.append(letter);
            } else {
                displayedWord.append("_");
            }
        }
        return displayedWord.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();

        String secretWord = chooseWord();
        StringBuilder guessedLetters = new StringBuilder();
        int attempts = 6;

        System.out.println("Welcome to Hangman!");
        System.out.println("Try to guess the secret word.");
        System.out.println(displayWord(secretWord, guessedLetters.toString()));

        while (true) {
            System.out.print("Guess a letter: ");
            String guess = scanner.nextLine().toLowerCase();

            if (guess.length() != 1 || !Character.isLetter(guess.charAt(0))) {
                System.out.println("Please enter a single letter.");
                continue;
            }

            if (guessedLetters.toString().contains(guess)) {
                System.out.println("You've already guessed that letter.");
                continue;
            }

            guessedLetters.append(guess);

            if (secretWord.contains(guess)) {
                System.out.println("Correct!");
            } else {
                System.out.println("Incorrect!");
                attempts--;
            }

            String displayedWord = displayWord(secretWord, guessedLetters.toString());
            System.out.println(displayedWord);

            if (!displayedWord.contains("_")) {
                System.out.println("Congratulations! You've guessed the word: " + secretWord);
                break;
            }

            if (attempts == 0) {
                System.out.println("You've run out of attempts! The secret word was: " + secretWord);
                break;
            }

            System.out.println("Attempts left: " + attempts);
        }
    }
}
